RELEASE/scripts/autoscend/combat/auto_combat_default_stage4.ash

```diff
@@ -12,46 +12,17 @@ string auto_combatDefaultStage4(int round, monster enemy, string text)
 	retval = auto_combatLicenseToAdventureStage4(round, enemy, text);
 	if(retval != "") return retval;
 	
-	//olfaction is used to spawn 2 more copies of the target at current location.
-	//as well as eliminate the special rule that reduces the odds of encountering the same enemy twice in a row.
-	if(auto_wantToSniff(enemy, my_location()))
+	//sniffers are skills that increase the odds of encountering this same monster again in the current zone.
+	skill sniffer = getSniffer(enemy);
+	if(!contains_text(combatState, "sniffed") && !isSniffed(enemy) && auto_wantToSniff(enemy, my_location()) && sniffer != $skill[none])
 	{
-		if(canUse($skill[Transcendent Olfaction]) && (have_effect($effect[On The Trail]) == 0))
+		if(sniffer == $skill[Perceive Soul])		//mafia does not track the target of this skill so we must do so.
 		{
-			handleTracker(enemy, $skill[Transcendent Olfaction], "auto_sniffs");
-			return useSkill($skill[Transcendent Olfaction]);
-		}
-
-		if(canUse($skill[Make Friends]) && get_property("makeFriendsMonster") != enemy && my_audience() >= 20)
-		{
-			handleTracker(enemy, $skill[Make Friends], "auto_sniffs");
-			return useSkill($skill[Make Friends]);
-		}
-
-		if(!contains_text(get_property("longConMonster"), enemy) && canUse($skill[Long Con]) && (get_property("_longConUsed").to_int() < 5))
-		{
-			handleTracker(enemy, $skill[Long Con], "auto_sniffs");
-			return useSkill($skill[Long Con]);
-		}
-
-		if(canUse($skill[Perceive Soul]) && enemy != get_property("auto_bat_soulmonster").to_monster())
-		{
-			handleTracker(enemy, $skill[Perceive Soul], "auto_sniffs");
 			set_property("auto_bat_soulmonster", enemy);
```

Comment:
Perceive Soul only lasts for 30 turns. We should track when this will expire. If in same zone for >30 turns, may not have monster sniffed any longer even though we expect it to be. I don't believe this will be a common issue so will leave up to you if you want to fix now or create a github issue so we don't forget about it.

Reply:
Yea, it needs better tracking. Ideally it should be tracked via mafia.